hivemind/averaging/matchmaking.py

```diff
@@ -166,26 +175,25 @@ async def _request_join_potential_leaders(self, timeout: Optional[float]) -> Gro
                         self.assembled_group.set_exception(e)
                     raise e
 
-    async def request_join_group(self, leader: PeerID, expiration_time: DHTExpiration) -> Optional[GroupInfo]:
+    async def request_join_group(self, leader: PeerID) -> Optional[GroupInfo]:
```

Comment:
Can we make this method a private one?

Reply:
We can, but i'd rather not: it would cause dissonance with get_request_expiration_time, rpc_join_group, etc

Reply:
Not sure about the dissonance: we have methods that are intended to be parts of the class public interface and those that are only meant to be used internally; the naming distinction simply helps reduce the interaction surface

Reply:
There are just as "private" (or more private) methods such as get_request_expiration_time rpc_join_group

We won't be able to rename rpc_join_group